spec/javascripts/stimulus/enterprise_form_controller_test.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { Application } from "stimulus";
+import side_menu_controller from "../../../app/webpacker/controllers/side_menu_controller";
```

Comment:
I'm wondering if there is any way to test `enterprise_form_controller` controller without `side_menu_controller` ?

 - If yes, I think it should be done. I don't think we need to click on element, but only send event.
 - If no, I think `enterprise_form_controller` should be included to `side_menu_controller`.

What. do you think?

Reply:
I don't think it's possible to test `enterprise_form` in isolation because the payload for `show() { detail: { currentActiveTab, newActiveTab } }` is unique.
I did think of making these into a single controller, but I felt it was easier to understand what was happening when there were two of them working together. Especially because the elements that they are concerned with (side menu handles `tabs` and enterprise form handles `form`) are different.
Also, I couldn't think of an appropriate name for such a controller. 
Should I try to make them into a single controller? I think this takes a hit on the readability of the code though.

Reply:
I re-read your code, trying to understand better how elements are interacting. 
I'm now more confident that (IMHO) those two controllers should be merged into one, for example `panel_controller` which handle at the same time the side menu on the left (which could be on top adctually, so I'd rename it into menu?) and the content that are displayed or not, strongly depending on what is selected/clicked on the menu.
To my opinion, it'll be easier to understand (no event between controllers will be sent, I think it's easier to understand), and will be fully generic (ie. not depending on any enterprise-something).


What do you think?

Reply:
oh I see what you mean. I think you're also thinking about the menu on the top `Dashboard Products ...` and that makes sense (maybe we could use this controller to render those thing when we get there). Ok I'll convert it into a single `menu_controller`. Thanks!

Reply:
Yes, It can be used somewhere else (one day?!) and it should be generic. 
I though about `panel` because `menu` makes me think that we don't talk about content (but only about clickable links). 
Maybe it's better to call it `tabs_and_panels_controller`? I don't know, I'm very bad about naming something...

Anyway, thanks a lot for your comprehension, and your patience!

Reply:
Could you take a look jibbes? Sorry, it took me sometime to get this done. I always mess something up when I need to go back and do a rebase, so it takes a couple of tries to get it right! :tired_face: